specification/securityinsights/resource-manager/Microsoft.SecurityInsights/preview/2022-01-01-preview/MitreCoverage.json

```diff
@@ -0,0 +1,167 @@
+      "get": {
+        "tags": [
+          "MitreCoverage"
+        ],
+        "description": "Fetches Products Mitre Coverage from Data sources",
+        "operationId": "Get",
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/1.0/types.json#/parameters/WorkspaceName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Products Mitre Coverages fetched",
+            "schema": {
+              "$ref": "#/definitions/ProductsMitreCoverageList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed."
+          }
+        },
+        "x-ms-examples": {
+          "Get": {
+            "$ref": "./examples/MitreCoverage/Get.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProductMitreCoverageModel": {
+      "description": "A wrapper to the resource of type T, which contains Arm's first level fields such as id, name and type, with the resource properties",
+      "required": [
+        "properties"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../common/1.0/types.json#/definitions/ResourceWithEtag"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Gets or sets nested level of properties which contains the resource content",
+          "type": "object",
+          "allOf": [
+            {
+              "$ref": "#/definitions/ProductMitreCoverageProperties"
+            }
+          ],
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "ProductMitreCoverageProperties": {
+      "type": "object",
+      "properties": {
+        "productName": {
+          "description": "Gets Or sets the product name that covers a specific tactic and technique.",
+          "type": "string",
+          "readOnly": true,
+          "example": "MDATP"
+        },
+        "vendorName": {
+          "description": "Gets Or sets the vendor of the Product.",
+          "type": "string",
+          "readOnly": true,
+          "example": "Microsoft"
+        },
+        "coverageSize": {
+          "format": "int32",
+          "description": "Gets Or sets the Coverage Size of the product for a specific tactic and technique.",
+          "type": "integer",
+          "readOnly": true,
+          "example": 2
+        },
+        "technique": {
```

Comment:
>technique [](http://example.com/codeflow?start=9&length=9)

If this is selected from a set of known techniques, you should model it as a string enum.

Reply:
the list may vary, and can be changed.
i would really avoid coupling my api to a set of closed (snapshot) values.
otherwise i should create a new api for every change.

Reply:
If you use modelAsString: true, you can add enum values without an API version change. I won't block on this, but if it's a relatively static set of values, enum is preferred.